### PR TITLE
Adds promotions_muted_at User field

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request adds a new `promotions_muted_at` field to our User model and adds an admin form to set it to the current time (for testing). The next PR (#1124) will modify our User observer to delete the user's Customer.io profile if this field is set, but holding off that, for now, to hopefully keep this easy to review.

### How should this be reviewed?

👀 

### Any background context you want to provide?

It'd be sweet to get the `data-confirm` code from Aurora working in this app to ask the staff member to confirm before muting (or sending a password reset), but out of scope for this PR.

### Relevant tickets

References [Pivotal #176771234](https://www.pivotaltracker.com/story/show/176771234).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

### CODE REVIEW

File: app/Http/Controllers/Web/Admin/UserController.php [hunk 1]
```diff
@@ -154,4 +155,24 @@ public function sendPasswordReset(User $user, Request $request)
...
             ->with('flash', 'Sent ' . $type . ' email to user.');
     }
+
+    /**
+     * Mutes promotions for user.
+     *
+     * @param User $user
+     * @return \Illuminate\Http\Response
+     */
+    public function mutePromotions(User $user)
```
Comment (DFurnes) [inline app/Http/Controllers/Web/Admin/UserController.php:165]: This is subjective, but I think I'd lean towards making a new `Admin\PromotionsController` for this, and putting this code under the `destroy` method (following [this old DHH advice](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/)). It keeps our controllers from getting too long, and kinda elegantly mirrors the route structure. (And if we ever wanted  a resubscribe button, it'd be easy to put in `PromotionsController@store`!)

File: routes/web.php [hunk 2]
```diff
@@ -102,6 +102,10 @@
                 'as' => 'users.resets.create',
                 'uses' => 'Admin\UserController@sendPasswordReset',
             ]);
+            Route::post('users/{user}/mute-promotions', [
+                'as' => 'users.mute-promotions',
+                'uses' => 'Admin\UserController@mutePromotions',
+            ]);
```
Comment (DFurnes) [inline routes/web.php:108]: Also subjective, but I like the idea of keeping this RESTful & considering this a `promotions` sub-resource on the user that we can delete (and following the same convention for the route alias):

```suggestion
            Route::delete('users/{user}/promotions', [
                'as' => 'users.promotions.destroy',
                'uses' => 'Admin\PromotionsController@destroy',
            ]);
```

File: resources/views/admin/users/partials/danger-zone.blade.php [hunk 3]
```diff
@@ -26,6 +26,25 @@
...
+            <div class="form-item">  
+                <label for="mute-promotions" class="field-label">Mute Promotions</label>
+
+                <p class="footnote">This will delete the user's Customer.io profile.</p>
+            </div>
+
+            <div class="form-actions">
+                <button type="submit" class="button -secondary">
+                    Mute promotions
+                </button>
```
Comment (DFurnes) [inline resources/views/admin/users/partials/danger-zone.blade.php:43]: Last thought – it'd be great to have a test for this form that asserts that the `muted_promotions_at` field is set & the Customer.io API request is fired! You can use [WebActionTest](https://github.com/DoSomething/northstar/blob/master/tests/Http/Web/WebActionTest.php) as an example for testing web routes.

File: app/Http/Controllers/Web/Admin/UserController.php [hunk 4]
```diff
@@ -154,4 +155,24 @@ public function sendPasswordReset(User $user, Request $request)
...
+        $this->authorize('delete', $user);
+
+        logger('Muting promotions', ['user' => $user->id]);
+
+        $user->promotions_muted_at = Carbon::now();
+        $user->save();
+
+        return redirect()
+            ->route('admin.users.show', $user->id)
+            ->with('flash', 'Promotions muted for user. Shhh....');
```
Comment (DFurnes) [inline app/Http/Controllers/Web/Admin/UserController.php:176]: This is great copy. 👏

**Review by DFurnes — APPROVED** [review]

File: resources/views/admin/users/partials/danger-zone.blade.php (hunk 3 above)
Comment (aaronschachter) [reply resources/views/admin/users/partials/danger-zone.blade.php:43]: There are some tests in place for the Customer.io API request introduced in #1124 -- I can add a new test class for the admin user form over in that PR.

**Pull request merged**